### ISSUE TITLE
[BUG] [Spring] [Spring-Boot] Exploded query parameters are not correctly serialized in controller

### ISSUE THREAD

**Issue by GregDThomas** [OpenAPITools/openapi-generator#14860]
#### Bug Report Checklist

- [X] Have you provided a full/minimal spec to reproduce the issue?
- [X] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [X] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [X] Have you searched for related issues/PRs?
- [X] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

When generating code with an exploded query parameter, the `spring` generator with the `spring-boot` library does not correctly resolve the supplied parameters. Some experimentation suggests that the serialization is using the name of the field in the Java object, not the name of the field in the OpenAPI specification.

##### openapi-generator version

6.4.0

##### OpenAPI declaration file content or url

https://github.com/GregDThomas/openapi-generator-spring-boot-example/blob/main/specs/explode-query-parameter.yaml

##### Generation Details

Using the OpenAPI Generator Gradle Plugin with 
```
    generatorName = "spring"
    library = 'spring-boot'
```

##### Steps to reproduce

See the sample project at https://github.com/GregDThomas/openapi-generator-spring-boot-example/blob/main/specs/explode-query-parameter.yaml and more specifically the tests in ExampleControllerTest

Specifically - 
- the fields in the exploded object that match the name of the field in the POJO (`someCamelString`) are correctly serialized.
- the fields in the exploded object that do no match the name of the field in the POJO are not correctly serialized.
-- `some-kebab-string` / `someKebabString`
-- `some_snake_string` / `someSnakeString`

**Comment by GregDThomas** [comment]
One solution is to "explode" the object in the method signature, so instead of
```java
    default ResponseEntity<SomeReturnValue> search(
        @NotNull @Parameter(name = "object-param", description = "", required = true, in = ParameterIn.QUERY) @Valid SearchObjectParamParameter objectParam,
...
```
becomes
```java
    default ResponseEntity<SomeReturnValue> search(
      @Parameter(name = "kebab-param", description = "", in = ParameterIn.QUERY) @Valid @RequestParam(value = "kebab-param", required = false) String kebabParam
      @Parameter(name = "snake_param", description = "", in = ParameterIn.QUERY) @Valid @RequestParam(value = "snake_param", required = false) String snakeParam
...

```
But don't know if this is the right approach for Spring Boot, or if there is a better one. Suggestions welcome.

**Comment by wing328** [comment]
thanks for reporting the issue.

cc @cachescrubber (2022/02) @welshm (2022/02) @MelleD (2022/02) @atextor (2022/02) @manedev79 (2022/02) @javisst (2022/02) @borsch (2022/02) @banlevente (2022/02) @Zomzog (2022/09)

**Comment by borsch** [comment]
Probably make sense to explode object into parameters. There is no may to map Object with `@RequestParam` with custom names .. Also would make sense to add configuration properties that would control this

**Comment by GregDThomas** [comment]
Yeah, I was coming to the conclusion that there was a sane way to map an Object to `@RequestParam` for Spring Boot, not sure if we need a config property to control the behaviour, but I guess there may be some edge cases (that I can't think of!) where the current behaviour is desirable. I'll see if I can get some time to look at this.

**Comment by borsch** [comment]
> I'll see if I can get some time to look at this.

@GregDThomas that's exactly the reason, why I would prefer to go with config option and by default leave as it is. I believe there are already users, which are fine with current behavior. Current behavior works pretty fine, when request parameters name has exactly same names as generate model's fields. Example

```yaml
/api/search:
    get:
      operationId: Search
      parameters:
        - name: regularParam
          in: query
          required: false
          schema:
            type: string
        - name: objectParam
          in: query
          required: true
          schema:
            type: object
            properties:
              someField1:
                type: string
              someField2:
                type: string
              someCamelString:
                type: string
      responses:
        '200':
          description: Some description.
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/SomeReturnValue'
```

and generated model would be

```java
public class ObjectParam {
  private String someField1;
  private String someField2;
  private String someCamelString;
}
```

In above case current generator's behavior is perfectly fine

**Comment by GregDThomas** [comment]
More research over lunch has turned up https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/method/support/HandlerMethodArgumentResolver.html which should allow a customer parameter annotation to be used - I may experiment with that to see if I can get it working ...

**Comment by borsch** [comment]
> More research over lunch has turned up https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/method/support/HandlerMethodArgumentResolver.html which should allow a customer parameter annotation to be used - I may experiment with that to see if I can get it working ...

I saw that thing. To be honest I would prefer to avoid this. Think about cases, when in single model you are trying to deal with list of custom models

```yaml
/api/search:
    get:
      operationId: Search
      parameters:
        - name: regularParam
          in: query
          required: false
          schema:
            type: string
        - name: objectParam
          in: query
          required: true
          schema:
            type: object
            properties:
              someField1:
                type: string
              listField:
                type: array
                items:
                  type: object
                  properties:
                    id:
                      type: integer
                    name:
                      type: string
                    statuses:
                      type: string
                      enum: [enum1, enum2, enum3]
      responses:
        '200':
          description: Some description.
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/SomeReturnValue'
```

`/api/search?listField[0].id=1&listField[0].name=asd&listField[0].statuses=enum1,enum2`


Handling such thing would be a real pain ..

What your thoughts on this?
@cachescrubber (2022/02) @welshm (2022/02) @MelleD (2022/02) @atextor (2022/02) @manedev79 (2022/02) @javisst (2022/02) @borsch (2022/02) @banlevente (2022/02) @Zomzog (2022/09)

**Comment by GregDThomas** [comment]
> `/api/search?listField[0].id=1&listField[0].name=asd&listField[0].statuses=enum1,enum2`

I suspect there's no good way to handle that, tbh.

But I think;
* leaving default behaviour to use the current `@RequestParam` 
* Supporting an optional `@NamedRequestParam` which is a _simple_ HandlerMethodArgumentResolver that provides a mapping of primitives to named fields in a POJO (i.e. essentially a slight improvement over the existing `@RequestParam` support)

gives 80% of the benefit of exploded query params for 20% of the effort.

**Comment by ketankhairnar** [comment]
@GregDThomas did you find a workaround for this? we're also facing this issue. In generated model - it also has annotation with correct case - snake_case here. It still expects `queryBy` instead of `query_by` in query parameters

```
  @JsonProperty("query_by")
  private String queryBy;

```

**Comment by GregDThomas** [comment]
No; AFAICT you'll either need your field names to be camel case, otherwise requires a change to the generator code which I've yet to get time to take a look at.
